skyportal/models/photometry.py

```diff
@@ -201,6 +214,145 @@ def e_mag(cls):
             else_=None,
         )
 
+    @property
+    def magref(self):
+        """
+        Reference magnitude, e.g.,
+        the mean magnitude of a variable source,
+        or the magnitude of a source before a transient started.
+        """
+        if self.ref_flux is not None and self.ref_flux > 0:
+            return -2.5 * np.log10(self.ref_flux) + PHOT_ZP
+        else:
+            return None
+
+    @hybrid_property
+    def e_magref(self):
+        """The error on the reference magnitude."""
+        if (
+            self.ref_flux is not None
+            and self.ref_flux > 0
+            and self.ref_fluxerr is not None
+            and self.ref_fluxerr > 0
+        ):
+            return (2.5 / np.log(10)) * (self.ref_fluxerr / self.ref_flux)
+        else:
+            return None
+
+    @magref.expression
+    def magref(cls):
+        """The magnitude of the photometry point in the AB system."""
+        return sa.case(
+            [
+                (
+                    sa.and_(
+                        cls.ref_flux != None,  # noqa: E711
+                        cls.ref_flux != 'NaN',
+                        cls.ref_flux > 0,
+                    ),
+                    -2.5 * sa.func.log(cls.ref_flux) + PHOT_ZP,
```

Comment:
is this log10 or log2?

Reply:
According to this it is indeed log10, but I'm glad we made the check: https://www.w3resource.com/PostgreSQL/log-function.php#:~:text=The%20PostgreSQL%20log()%20function,as%20specified%20in%20the%20argument.